yajsapi/executor/index.ts

```diff
@@ -419,90 +419,104 @@ export class Executor implements ComputationHistory {
       logger.debug("Stopped processing debit notes.");
     }
 
-    async function find_offers(): Promise<void> {
-      let _subscription: Subscription;
+    async function find_offers_for_subscription(subscription: Subscription): Promise<void> {
+      emit(new events.SubscriptionCreated({ sub_id: subscription.id() }));
+      let _proposals;
       try {
-        _subscription = await builder.subscribe(market_api);
+        _proposals = subscription.events(self._worker_cancellation_token);
       } catch (error) {
-        emit(new events.SubscriptionFailed({ reason: error }));
-        throw error;
+        emit(
+          new events.CollectFailed({
+            sub_id: subscription.id(),
+            reason: error,
+          })
+        );
```

Comment:
Shouldn't we exit this function when `subscription.events()` fails? That's what the `yapapi` version does.